server/src/db/model/currencyModel.js

```diff
@@ -0,0 +1,72 @@
+const mongoose = require('mongoose');
+const CurrencySchema = require('../schema/currencySchema');
+const CurrencyModel = mongoose.model('currency', CurrencySchema);
+const bacenValidator = require('../../utils/validators/bacenValidator');
+const { array } = require('joi');
+
+const currency = {
+
+  listCurrencies: async function (aggregate) {
+
+    if (aggregate) {
+      return await CurrencyModel.aggregate([
```

Comment:
não é necessário usar a keyword `await` quando se pretende retornar uma promise.

 ```suggestion
      return CurrencyModel.aggregate([
```